src/ln/channel.rs

```diff
@@ -441,12 +441,9 @@ impl Channel {
 	/// that we're rejecting the new channel.
 	pub fn new_from_req(fee_estimator: &FeeEstimator, chan_keys: ChannelKeys, their_node_id: PublicKey, msg: &msgs::OpenChannel, user_id: u64, announce_publicly: bool) -> Result<Channel, HandleError> {
 		// Check sanity of message fields:
-		if msg.funding_satoshis >= (1 << 24) {
+		if msg.funding_satoshis >= msgs::MAX_FUNDING_SATOSHIS {
 			return Err(HandleError{err: "funding value > 2^24", msg: Some(msgs::ErrorAction::DisconnectPeer{})});
 		}
-		if msg.funding_satoshis > 21000000 * 100000000 {
```

Comment:
I removed this condition because I thought it never was going to be true because `MAX_FUNDING_SATOSHIS` should never be more high than 2,1x10^15, the test I added checks that.